Regression: compiler panics in lowerOptionMatch after c772024

## Summary

Commit `c772024` ("fix(checker): widen match result arms to expected unions") introduces a regression in the AIR lowering phase: when a function contains a `match` on an `Option<Dynamic>` whose arm calls an extern function (e.g. `decode::run`), `ard build` panics with a nil pointer dereference in `air.(*lowerer).declareModuleCallExtern`.

Reverting to `ae4059f` builds the same code without issue.

## Crash

```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x2 addr=0x18 pc=0x...]

goroutine 1 [running]:
github.com/akonwi/ard/air.(*lowerer).declareModuleCallExtern(...)
	air/lower.go:931 +0x124
github.com/akonwi/ard/air.(*functionLowerer).lowerExpr(...)
	air/lower.go:2644 +0x1f50
github.com/akonwi/ard/air.(*functionLowerer).lowerExprWithExpectedRaw(...)
	air/lower.go:1644 +0x788
github.com/akonwi/ard/air.(*functionLowerer).lowerContextualExpr(...)
	air/lower.go:1572 +0x2c
github.com/akonwi/ard/air.(*functionLowerer).lowerBlockWithDefault(...)
	air/lower.go:1555 +0xf0
github.com/akonwi/ard/air.(*functionLowerer).lowerOptionMatch(...)
	air/lower.go:3203 +0x340
```

## Repro

In an existing Ard project that has `ard/decode` available, add a function like:

```ard
fn nested_name(obj: Dynamic, field: Str) Str {
  let nested = decode::run(obj, decode::field(field, decode::nullable(decode::dynamic)))
    .expect("Missing nested field")
  match nested {
    n => decode::run(n, decode::field("name", decode::string)).expect("Missing nested name"),
    _ => "",
  }
}
```

`ard check` passes. `ard build` panics during lowering.

(`linear-cli` has multiple existing functions with the same shape that work fine — e.g. `draw_card_row` in `commands/tui.ard` — so the trigger is some interaction with surrounding code, not the pattern itself. Removing the new function still crashes once it has been compiled.)

## Version

`ard version` → `dev` (HEAD at `c772024`).

--- a/compiler/go/backend_test.go
+++ b/compiler/go/backend_test.go
@@ -307,6 +307,41 @@ fn main() Str {
 	}
 }
 
+func TestBuildProgramLowersOptionMatchArmModuleExternCall(t *testing.T) {
+	dir := t.TempDir()
+	if err := os.WriteFile(filepath.Join(dir, "ard.toml"), []byte("name = \"demo\"\nard = \">= 0.1.0\"\n"), 0o644); err != nil {
+		t.Fatal(err)
+	}
+	mainPath := filepath.Join(dir, "main.ard")
+	if err := os.WriteFile(mainPath, []byte(`use ard/decode
+
+fn nested_name(obj: Dynamic, field: Str) Str {
+  let nested = decode::run(obj, decode::field(field, decode::nullable(decode::dynamic)))
+    .expect("Missing nested field")
+  match nested {
+    n => decode::run(n, decode::field("name", decode::string)).expect("Missing nested name"),
+    _ => "",
+  }
+}
+
+fn main() {}
+`), 0o644); err != nil {
+		t.Fatal(err)
+	}
+
+	loaded, err := frontend.LoadModule(mainPath, backend.TargetGo)
+	if err != nil {
+		t.Fatalf("load module: %v", err)
+	}
+	program, err := air.Lower(loaded.Module)
+	if err != nil {
+		t.Fatalf("lower: %v", err)
+	}
+	if _, err := BuildProgram(program, filepath.Join(dir, "app"), loaded.ProjectInfo); err != nil {
+		t.Fatalf("build: %v", err)
+	}
+}
+
 func TestBuildProgramImportsProjectFFIForExternTypesOnlyUsedAsTypes(t *testing.T) {
 	dir := t.TempDir()
 	if err := os.WriteFile(filepath.Join(dir, "ard.toml"), []byte("name = \"demo\"\nard = \">= 0.1.0\"\n"), 0o644); err != nil {

